manager/rbac_v2_manager.py

```diff
@@ -0,0 +1,362 @@
+            raise
+
+    def _get_default_workspace_id(self, org_id: str) -> str:
+        """Fetch default workspace ID from RBAC service using Kessel SDK authentication."""
+        if not self._auth_credentials:
+            raise Exception("Kessel authentication not configured")
+
+        rbac_workspaces_url = f"{self._config.rbac_url}/api/rbac/v2/workspaces/?type=default"
+
+        try:
+            auth_token, _ = self._auth_credentials.get_token()
+
+            response = requests.get(
+                rbac_workspaces_url,
+                headers={
+                    "Authorization": f"Bearer {auth_token}",
+                    "x-rh-rbac-org-id": org_id
+                },
+                timeout=self._config.rbac_timeout,
+            )
+            response.raise_for_status()
+
+            workspaces = response.json()["data"]
+            if workspaces:
+                return workspaces[0]["id"]
+            else:
+                raise Exception("No default workspace found")
+
+        except Exception as e:
+            logger.error("Failed to fetch default workspace: %s", e)
+            raise
+
+    def check_default_workspace_permission_for_update(self, permission: str, org_id: str, principal_id: str) -> bool:
+        """Check if principal has permission on the default workspace."""
+
+        workspace_id = self._get_default_workspace_id(org_id)
+
+        object_ref = ResourceReference(
+            resource_type="workspace",
+            resource_id=workspace_id,
+            reporter=ReporterReference(type="rbac")
+        )
+
+        subject_ref = SubjectReference(
+            resource=ResourceReference(
+                resource_type="principal",
+                resource_id=f"redhat/{principal_id}",
+                reporter=ReporterReference(type="rbac")
+            )
+        )
+
+        check_request = CheckRequest(
+            object=object_ref,
+            relation=permission,
+            subject=subject_ref
+        )
+
+        response = self._kessel_client.CheckForUpdate(check_request)
+        return response.allowed
+
+    def get_workspace_ids_for_permission(self, permission: str, principal_id: str) -> Set[str]:
+        """Get workspace IDs that the principal has the given permission for."""
+
+        object_type = RepresentationType(
+            resource_type="workspace",
+            reporter_type="rbac"
+        )
+
+        subject_ref = SubjectReference(
+            resource=ResourceReference(
+                resource_type="principal",
+                resource_id=f"redhat/{principal_id}",
+                reporter=ReporterReference(type="rbac")
+            )
+        )
+
+        list_request = StreamedListObjectsRequest(
+            object_type=object_type,
+            relation=permission,
+            subject=subject_ref
+        )
+
+        workspace_ids = set()
+        for response in self._kessel_client.StreamedListObjects(list_request):
```

Comment:
I nearly forgot about this, but `StreamedListObjects` is paginated. You will have to handle the pagination yourself.
I think the default is 10 elements per page, but can be wrong:

This is how I handled the pagination in ROS.
https://github.com/RedHatInsights/ros-backend/blob/9ba3779739b98424f528bd24486313e22320af59/ros/lib/kessel/kessel_client.py#L34